### PR TITLE
chore: upgrade Slate

### PR DESCRIPTION
**Pull request by hejtful**
- Upgrade @prezly/slate-lists, slate and slate-react
- Add slate-dom to dependencies to fix an error
- Remove @types/slate-react as it was unnecessary
- Add @types/is-hotkey as it became necessary 🤷 
- Make Slate-related versions fixed in package.json
- \+
- Fix failing e2e tests on staging

### CODE REVIEW

**Review by CodingDive — APPROVED** [review]
Works great on my end! Thank you ❤️ 
Just curious, were these the fuzzy tests before, or did they break as part of the upgrade?

**Comment by hejtful** [comment]
They're from the focus/hover redesign, I just noticed them on staging after merging 😁

**Review by LarsTheGlidingSquirrel — APPROVED** [review]
Thank you! Tested the text plugin a bit and found no issue regarding the slate upgrade.

File: e2e-tests/tests/430-multimedia-plugin.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ Scenario('Multimedia plugin toolbar controls', async ({ I }) => {
     'aria-selected': 'true',
   })
   I.click('$modal-close-button')
-  I.seeElement('.media-wrapper.mobile\\:w-1\\/4')
+  I.seeElement('.media-wrapper.mobile\\:basis-1\\/4')
```
Comment (CodingDive) [inline e2e-tests/tests/430-multimedia-plugin.ts:46]: Would be nice to replace this with data-qa attribute

Comment (hejtful) [reply e2e-tests/tests/430-multimedia-plugin.ts:46]: I wouldn't agree in this case. This line is checking if the "quarter width media" option has been set correctly, and the best way to check that is by checking if the `.media-wrapper` element has the correct class.

**Pull request merged**